cmd/restic/cmd_backup.go

```diff
@@ -710,7 +713,21 @@ func runBackup(opts BackupOptions, gopts GlobalOptions, term *termstatus.Termina
 	if !gopts.JSON && !opts.DryRun {
 		progressPrinter.P("snapshot %s saved\n", id.Str())
 	}
-	if !success {
+	if opts.SummaryFilename != "" {
+		sf, err := os.OpenFile(opts.SummaryFilename, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0666)
```

Comment:
I'd prefer to extract that code into a separate function. `runBackup` is already far too long.